src/auth/src/errors.rs

```diff
@@ -12,57 +12,58 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-type BoxError = Box<dyn std::error::Error + Send + Sync>;
+use std::error::Error;
+use std::fmt::{Display, Formatter, Result};
 
+type BoxError = Box<dyn Error + Send + Sync>;
+
+/// Represents an auth error.  This error type indicates issues
+/// encountered during the auth process.
 #[derive(Debug)]
 pub struct AuthError {
+    /// A boolean value indicating whether the error is retryable. If true,
+    /// the operation that resulted in this error might succeed upon retry.
     is_retryable: bool,
+
+    /// The underlying source of the error. This provides more specific
+    /// information about the cause of the auth failure.
     source: BoxError,
 }
 
 impl AuthError {
+    /// Creates a new `AuthError`.
+    ///
+    /// # Arguments
+    /// * `is_retryable` - A boolean indicating whether the error is retryable.
+    /// * `source` - The underlying error that caused the auth failure.
+    ///
+    /// # Returns
+    /// A new `AuthError` instance.
     pub fn new(is_retryable: bool, source: BoxError) -> Self {
         AuthError {
             is_retryable,
             source,
         }
     }
 
+    /// Returns `true` if the error is retryable; otherwise returns `false`.
     pub fn is_retryable(&self) -> bool {
         self.is_retryable
     }
 }
 
-impl std::error::Error for AuthError {}
+impl Error for AuthError {}
```

Comment:
If we bother to capture the source error, we should make it available via the standard trait?

```suggestion
impl Error for AuthError {
  fn source(&self) -> Option<Box<dyn Error + 'static>> {
    Some(self.source)
  }
}
```